doc/build/sysbuild/index.rst

```diff
@@ -0,0 +1,370 @@
+enable MCUboot and build and flash the sample, like this:
+
+.. zephyr-app-commands::
+   :tool: west
+   :app: samples/hello_world
+   :board: reel_board
+   :goals: flash
+   :west-args: --sysbuild
+   :gen-args: -DSB_CONFIG_BOOTLOADER_MCUBOOT=y
+   :compact:
+
+this will build ``hello_world`` and ``mcuboot`` for the ``reel_board``, and then
+flash both ``mcuboot`` and ``hello_world``.
+
+More detailed information regarding the use of MCUboot with Zephyr  can be found
+in the `MCUboot with Zephyr`_ documentation page on the MCUboot website.
+
+.. note::
+   MCUBoot default configuration will perform a full chip erase when flashed.
+   This can be controlled through the MCUBoot ``ZEPHYR_TRY_MASS_ERASE`` Kconfig
+   setting. If this setting is on, then flashing only MCUBoot, for example using
+   ``west flash --domain mcuboot``, may erase the complete flash.
+
+Sysbuild configuration file
+***************************
+
+Sysbuild supports its own configuration file infrastructure similar to Zephyr
+build system.
+
+Sysbuild default looks for a configuration file named ``sysbuild.conf`` in the
+application top-level directory, like this:
+
+.. code-block:: none
+
+   <home>/application
+   ├── CMakeLists.txt
+   ├── prj.conf
+   └── sysbuild.conf
+
+A ``sysbuild.conf`` with the following assignment will enable ``mcuboot`` for
+the application, whenever sysbuild is used for building.
+
+.. code-block:: none
+
+   SB_CONFIG_BOOTLOADER_MCUBOOT=y
+
+Sysbuild also supports custom configuration file names with the
+``-DSB_CONF_FILE=<sysbuild-conf-file>`` CMake build setting.
+
+For example this allows you to create ``sysbuild-mcuboot.conf`` and then specify
+this file instead when building with sysbuild, as:
+
+.. zephyr-app-commands::
+   :tool: west
+   :app: samples/hello_world
+   :board: reel_board
+   :goals: flash
+   :west-args: --sysbuild
+   :gen-args: -DSB_CONF_FILE=sysbuild-mcuboot.conf
+   :compact:
+
+.. _sysbuild_zephyr_application:
+
+Adding Zephyr based application
+*******************************
+
+A board or an application might support additional Zephyr based applications.
+
+Supporting an extra Zephyr based application can be done with a single CMake
+function call in a :file:`CMakeLists.txt` or :file:`*.cmake` file.
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Add(
+     APPLICATION my_sample
+     SOURCE_DIR <path-to>/my_sample
+     BUILD_ALWAYS True
+   )
+
+this will include ``my_sample`` as an extra image in sysbuild, and using the
+same board as the main image.
+
+It is also possible to change the board used for the extra image, for example
+if the extra image should target a secondary core on the same SoC.
+
+For example, if a sample is built for board ``mps2_an521`` and the extra
+application must target the ``mps2_an521_remote`` board (cpu1), then ``BOARD``
+can be specified as part of the ``ExternalZephyrProject_Add()`` call:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Add(
```

Comment:
Question on this part:
Is there a command line alternative to this CMake directive ?

Also, does it allow the use of west debug on the secondary target ?

Reply:
> Question on this part: Is there a command line alternative to this CMake directive ?
> 

you mean like:
```
cmake ... -D_APP_DIR=samples/hello_world -DEXTRA_ZEPHYR_PROJECTS=<path>/sample/basic/blinky 
```
?

If this is what you mean, then not as of today, but you may suggest such feature here: https://github.com/zephyrproject-rtos/zephyr/issues/41640
but please provide a bit more details on the exact use-case.
Especially it is not clear how one would provide info as to board, toolchain, etc if needed.

> Also, does it allow the use of west debug on the secondary target ?

yes.
Described here:
https://github.com/zephyrproject-rtos/zephyr/blob/1d20fd5a10d201cf2c5fc20f4259121fcf25d468/doc/build/sysbuild/index.rst#sysbuild-debugging-west-debug

> To debug a different domain (Zephyr application), such as mcuboot, the --domain argument must be used, for example:
> 
> `west debug --domain mcuboot`

and here:
https://github.com/zephyrproject-rtos/zephyr/blob/1d20fd5a10d201cf2c5fc20f4259121fcf25d468/doc/develop/west/build-flash-debug.rst#multi-domain-debugging

Reply:
Actually, I was thinking of a multi-core use case, rather than combined-app.
Or does `--domain` allows to specify a domain dedicated hw target ?

Reply:
>  Or does --domain allows to specify a domain dedicated hw target ?

yes, if sysbuild contains builds for different cores (boards) then `--domain` will actually target the core (board) for which that build is targeting.

For followup work, see: https://github.com/tejlmand/zephyr/commit/efa77ac2e8507fea7bf4be4a67bd0b42af7899a7
where you'll see that two different cores are handled by sysbuild.

Reply:
Great, I'll have a look, thanks!